src/bidiMapper/modules/network/NetworkStorage.ts

```diff
@@ -271,25 +256,47 @@ export class NetworkStorage {
       );
     }
 
-    const request = this.getRequestById(params.request)!;
+    const request = this.getRequestById(params.request);
     if (request === undefined) {
-      throw new NoSuchNetworkDataException(
-        `No data for request ${params.request}`,
+      throw new NoSuchNetworkDataException(`No data for ${params.request}`);
+    }
+
+    let result: Network.GetDataResult | undefined = undefined;
+    switch (params.dataType) {
+      case Network.DataType.Response:
+        result = await this.#getCollectedResponseData(request);
+        break;
+      case Network.DataType.Request:
+        result = await this.#getCollectedRequestData(request);
+        break;
+      default:
+        throw new UnsupportedOperationException(
+          `Unsupported data type ${params.dataType}`,
+        );
+    }
+
+    if (params.disown && params.collector !== undefined) {
+      this.#collectorsStorage.disownData(
+        request.id,
+        params.dataType,
+        params.collector,
       );
+      // `disposeRequest` disposes request only if no other collectors for it are left.
```

Comment:
Nit: I would prefer this be a TSDoc comment on the method, or at least copy it there as well

Reply:
added a JSDOC on `disposeRequest`